src/dune_rules/lib.ml

```diff
@@ -396,10 +431,11 @@ end
 
 type db =
   { parent : db option
-  ; resolve : Lib_name.t -> resolve_result Memo.t
+  ; resolve : Lib_name.t -> resolve_result_with_multiple_results Memo.t
+  ; resolve_library_id : Lib_info.Library_id.t -> resolve_result Memo.t
   ; instantiate :
       (Lib_name.t -> Path.t Lib_info.t -> hidden:string option -> Status.t Memo.t) Lazy.t
-  ; all : Lib_name.t list Memo.Lazy.t
+  ; all : Lib_info.Library_id.t list Memo.Lazy.t
```

Comment:
Is this a correct use of id's? we aren't supposed to list disabled libraries here

Reply:
thanks for catching that. I changed it back to return a Lib_name.t list